core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -71,31 +76,70 @@
 
+  /**
+   * Restores a page store a the configured location, updating meta store and evictor.
+   *
+   * @param pageStore page store
+   * @param options page store options
+   * @param metaStore meta store
+   * @param evictor evictor
+   * @return whether the restore succeeds or not
+   */
+  private static boolean restore(
+      PageStore pageStore, PageStoreOptions options, MetaStore metaStore, CacheEvictor evictor) {
+    LOG.info("Restore PageStore with {}", options);
+    Path rootDir = Paths.get(options.getRootDir());
+    if (!Files.exists(rootDir)) {
+      LOG.error("Directory {} does not exist", rootDir);
+      return false;
+    }
+    try (Stream<PageInfo> stream = pageStore.getPages()) {
+      Iterator<PageInfo> iterator = stream.iterator();
+      while (iterator.hasNext()) {
+        PageInfo pageInfo = iterator.next();
+        if (pageInfo == null) {
+          LOG.error("Invalid page info");
+          return false;
+        }
+        metaStore.addPage(pageInfo.getPageId(), pageInfo);
+        evictor.updateOnPut(pageInfo.getPageId());
+        if (metaStore.bytes() > pageStore.getCacheSize()) {
+          LOG.error("Loaded pages exceed cache capacity ({} bytes)",
+                  pageStore.getCacheSize());
+          return false;
+        }
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to restore PageStore", e);
+      return false;
+    }
+    LOG.info("Restored PageStore with {} existing pages and {} bytes",
+        metaStore.pages(), metaStore.bytes());
+    return true;
+  }
+
   /**
    * @param conf the Alluxio configuration
    * @return an instance of {@link LocalCacheManager}
    */
   public static LocalCacheManager create(AlluxioConfiguration conf) throws IOException {
     MetaStore metaStore = MetaStore.create();
     CacheEvictor evictor = CacheEvictor.create(conf);
-    PageStore pageStore = PageStore.create(conf);
+    PageStoreOptions options = PageStoreOptions.create(conf);
+    PageStore pageStore = null;
+    boolean restored = false;
     try {
-      Collection<PageInfo> pageInfos = pageStore.getPages();
-      LOG.info("Creating LocalCacheManager with {} existing pages", pageInfos.size());
-      for (PageInfo pageInfo : pageInfos) {
-        PageId pageId = pageInfo.getPageId();
-        metaStore.addPage(pageId, pageInfo);
-        evictor.updateOnPut(pageId);
-      }
-      return new LocalCacheManager(conf, metaStore, pageStore, evictor);
+      pageStore = PageStore.create(options, false);
+      restored = restore(pageStore, options, metaStore, evictor);
     } catch (Exception e) {
-      try {
-        pageStore.close();
-      } catch (Exception ex) {
-        e.addSuppressed(ex);
-      }
-      throw new IOException("failed to create local cache manager", e);
+      LOG.error("Failed to restore PageStore", e);
+    }
+    if (!restored) {
+      metaStore.reset();
+      evictor.reset();
+      pageStore = PageStore.create(options, true);
```

Comment:
Should the previous page store be closed before we create a new one?

Reply:
added